### PR TITLE
feat: use new restricted liveness probe endpoint

### PR DESCRIPTION
**Pull request by andyzhangx**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
feat: use new restricted liveness probe endpoint

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
feat: use new restricted liveness probe endpoint
```

### CODE REVIEW

**Comment by sushain97** [comment]
Just an FYI that upgrading my chart to v1.14.0 to include this change resulted in it the deployment crashlooping due to a failing healthcheck. I've downgraded to v1.13.0 and things are healthy again.

**Comment by andyzhangx** [comment]
> Just an FYI that upgrading my chart to v1.14.0 to include this change resulted in it the deployment crashlooping due to a failing healthcheck. I've downgraded to v1.13.0 and things are healthy again.

@sushain97 is it failing on linux node or windows node?

**Comment by sushain97** [comment]
Linux x86_64

**Comment by andyzhangx** [comment]
@sushain97 which pod is crashing? controller pod or daemonset pod?

File: charts/v1.14.0/csi-driver-smb/templates/csi-smb-controller.yaml [hunk 1]
```diff
@@ -106,17 +106,15 @@ spec:
...
             - containerPort: {{ .Values.controller.metricsPort }}
               name: metrics
               protocol: TCP
           livenessProbe:
             failureThreshold: 5
             httpGet:
+              host: localhost
               path: /healthz
-              port: healthz
+              port: {{ .Values.node.livenessProbe.healthPort }}
```
Comment (andyzhangx) [inline charts/v1.14.0/csi-driver-smb/templates/csi-smb-controller.yaml:117]: @sushain97 I found the bug!

Comment (andyzhangx) [reply charts/v1.14.0/csi-driver-smb/templates/csi-smb-controller.yaml:117]: stupid copy & paste bug

**Comment by andyzhangx** [comment]
> @sushain97 which pod is crashing? controller pod or daemonset pod?

@sushain97 would be fixed by https://github.com/kubernetes-csi/csi-driver-smb/pull/769

**Comment by sushain97** [comment]
> > @sushain97 which pod is crashing? controller pod or daemonset pod?
> 
> @sushain97 would be fixed by #769

Thanks! :)

FWIW, I'm a little confused by the versioning here. I assumed that v1.14.0 wouldn't change after Jan 16 (https://github.com/kubernetes-csi/csi-driver-smb/releases) but it appears that it's being changed even in #769? I would have expected a v1.14.1.

**Comment by andyzhangx** [comment]
@sushain97 v1.14.0 image won't change, but I changed the helm chart config a little to not expose one port on the node

**Comment by andyzhangx** [comment]
@sushain97 PR merged, could you run `helm repo update` and then install v1.14.0 chart again, thanks.

**Comment by sushain97** [comment]
Looks like it works now! Thanks for the speedy fix.

**Pull request merged**